modules/library/main/src/main/java/org/geotools/feature/AttributeTypeBuilder.java

```diff
@@ -197,9 +198,7 @@
 	 *
 	 */
     public AttributeTypeBuilder() {
-		this( new FeatureTypeFactoryImpl() );
-		init();
```

Comment:
Btw, removing this init means the userData map won't be initialized to a empty map, calling addUserData right after the constructor seems like it will generate a NPE (unless I'm missing something, just had another quick look)